apis/python/src/tiledbsoma/tiledb_object.py

```diff
@@ -20,6 +26,11 @@ class TileDBObject(ABC, somacore.SOMAObject):
     _context: SOMATileDBContext
     _metadata: MetadataMapping
 
+    # "r", "w", or falsey
+    _open_mode: str = ""
```

Comment:
tiny nit:  can we type "mode" throughout, e.g., `Optional[Literal["r", "w"]]`?  It would help annotate the API upstream, e.g. in the `open()` method.

Reply:
More than I nit, I think this is important.

* In the open, we can default an _argument_ which is `None` to be `"r"` -- as in Python's `open(file, mode="r", ...)`, or `tiledb.open` which also defaults `mode` to `"r"`
* But (here) we store always either`"r"` or `"w"` as an attribute

Reply:
Added `OpenMode = Literal["r","w"]`, now the internal attribute is `Optional[OpenMode]` and the open() argument is `mode: OpenMode`. I don't see appeal in separating them, since there are three and not four valid states.

Reply:
Works for me.  I also noticed that @thetorpedodog made the same change upstream in the SOMA ABC, so things are aligned.